src/Utility.js

```diff
@@ -5,12 +5,14 @@ class Utility {
     this.singleUserData = this.dataSet.filter(stat => stat.userID === this.userID);
   }
 
-  returnWeekDatesOnly(week) {
-    return [...this.singleUserData].splice(-7 * week, 7).map(day => day.date);
+  returnWeekDatesOnly(date) {
+    let findIndexForDate = this.singleUserData.findIndex(stat => stat.date === date);
+    return [...this.singleUserData].splice(findIndexForDate - 6, 7).map(day => day.date);
```

Comment:
In this situation, we're splicing from the array of objects within this.singleUserData which (I think) mutates that original array. So if we run returnWeekDatesOnly, does that remove the 7 objects from the this.singleUserData property permanently? And will later methods that rely on that same array stored in that property be affected?

Reply:
@Dunn-Austen Great question!  In my fitlit we used `slice` instead of `splice` for that exact reason.   

However, in this project, they are using the spread operator before they `splice`. This means that a copy of the original array is being created and subsequently `spliced` so the original array is not mutated.   

I thought about rewriting these with slice but tried to challenge myself to work from their code as much as possible rather than replacing their working code with something that made more sense to me personally.

Reply:
Thank you for so promptly clarifying. The spread operator nuance had escaped me. 
All approved from my side. I'll get this merged ASAP.